docpages/advanced_reference/separate-events.md

```diff
@@ -0,0 +1,68 @@
+\page separate-events Separating events into new classes
+
+Now, if you're someone that loves file organisation (or you hate how cluttered your `main.cpp` has become) then you may be interested in moving events into separate classes outside of the `main.cpp` file. This is a great way to improve readability and can be helpful in many cases! For example, you can have two classes on the same event, except one could be reading messages for spam and one could be reading messages for bad words!
+
+In this tutorial, we'll be taking the [Listening to messages](/detecting-messages.html) example and moving the `on_message_create` event into a different class.
+
+To get started, you can create a folder called `listeners` inside `src` (where your `main.cpp` is) if you'd like to put it there! We'll be doing exactly that so, if you'd like to stick along with the tutorial, get creating that folder!
+
+Now, you can create a new header file in this folder! You don't need the cpp file as you can still write code in the header so, if your IDE creates a `.cpp` file, feel free to delete it! For this tutorial, we'll be naming this file `message_listener`!
+
+If you're using CMake, you'll need to add this to your `CMakeLists.txt`. Some IDEs automatically do this but it's always worth double-checking!
+
+Once that's done, it should look similar to this (this screenshot has more files in, so it won't be identical!):
+
+\image html file_example_listeners.png
+
+Now, let's go ahead and write some code in this `message_listener.h`.
+
+~~~~~~~~~~{.cpp}
+#pragma once
```

Comment:
putting non-inline code into a header like this is kind of a no-no.
Can this be redone to have a separate .cpp and .h file where the .h has the forward declaration only?
We would be teaching very bad habits to make them do it this way.

Reply:
Yep, we can do this! I didn't realise this was bad practice so that's good to know.